detox/android/detox/src/full/java/com/wix/detox/DetoxManager.java

```diff
@@ -85,12 +96,12 @@ public void run() {
     boolean stopping = false;
 
     void stop() {
-        Log.i(LOG_TAG, "Stopping Detox.");
-        handler.postAtFrontOfQueue(new Runnable() {
+        Log.i(LOG_TAG, "Stopping Detox!");
+        executor.execute(new Runnable() {
             @Override
             public void run() {
                 if (stopping) return;
-                stopping = true;
+                stopping = true; // TODO is this needed?
```

Comment:
Not sure either, but better to be safe than sorry. 😆

Reply:
Also, the question is, if both jobs are busy (you have 2 max), does that mean that `stop()` will have to wait quite a bit? Just wondering...

Reply:
No, it should be able to close things abruptly

Reply:
@d4vidi, it is going to close things abruptly ONLY in case if `currentStatus` query execution is rapid. Otherwise, if your `invoke` is stuck, and then the app sends `currentStatus` for the clarification, your thread pool already is full (2 out of 2).

Probably, there's no way `currentStatus` can get stuck but just saying. 🤷

Reply:
Oh I see what you mean. Yes, `currentStatus` should finish up promptly. Are you suggesting to increase the pool to 3? That can be something to consider indeed.

Reply:
Oh, I'm not entirely sure if `pool=3` configuration is a safer thing to introduce. Let's leave it as is, but I recommend adding a TODO in the code at least so it is highlighted as a potential issue.

Reply:
On 2nd though, is it really bad if `stop` takes a while to complete?

Reply:
Well, the Node.js side can just kill the app in that case, indeed. So you are probably right.